### PR TITLE
MapPath dangerous methods

### PR DESCRIPTION
**Pull request by cbortos-d2l**

### CODE REVIEW

File: src/D2L.CodeStyle.Analyzers/D2L.CodeStyle.Analyzers.csproj [hunk 1]
```diff
@@ -150,6 +150,7 @@
       <HintPath>..\..\packages\System.Reflection.Metadata.1.2.0\lib\portable-net45+win8\System.Reflection.Metadata.dll</HintPath>
       <Private>True</Private>
     </Reference>
+    <Reference Include="System.Web" />
```
Comment (j3parker) [inline src/D2L.CodeStyle.Analyzers/D2L.CodeStyle.Analyzers.csproj]: Hm, could we go with strings rather than referencing this?

**Comment by j3parker** [comment]
Sweet, cool stuff!

**Comment by j3parker** [comment]
(One of the reasons to not reference `System.Web`: it would prevent us from shipping a netstandard/.NET core build... we don't currently but it's definitely something we'll want to do in the future..)

File: tests/D2L.CodeStyle.Analyzers.Test/SpecRunner.cs [hunk 2]
```diff
@@ -350,6 +351,12 @@ ImmutableArray<string> arguments
 					MetadataReference
 						.CreateFromFile( typeof( Enumerable ).Assembly.Location ) )
 
+				// system.web
```
[the remark this answers is not in the document]
Comment (cbortos-d2l) [reply tests/D2L.CodeStyle.Analyzers.Test/SpecRunner.cs]: The `spec` test code references classes from this assembly (i.e. `System.Web.Hosting.HostingEnvironment`). Without it, the symbols were `null`

[the remark this answers is not in the document]
Comment (cbortos-d2l) [reply tests/D2L.CodeStyle.Analyzers.Test/SpecRunner.cs]: Oh, I see. I can do that.

File: src/D2L.CodeStyle.Analyzers/D2L.CodeStyle.Analyzers.csproj (hunk 1 above)
Comment (j3parker) [reply src/D2L.CodeStyle.Analyzers/D2L.CodeStyle.Analyzers.csproj]: Yeah :(

File: tests/D2L.CodeStyle.Analyzers.Test/SpecRunner.cs [hunk 3]
```diff
@@ -100,7 +100,7 @@ private sealed class DiagnosticComparer : IEqualityComparer<Diagnostic> {
 				.Intersect(
 					m_expectedDiagnostics,
 					DiagnosticComparer.Instance
-				).ToImmutableHashSet();
```
Comment (j3parker) [inline tests/D2L.CodeStyle.Analyzers.Test/SpecRunner.cs]: D'oh, thanks!

**Comment by j3parker** [comment]
@cbortos-d2l isn't sure if he'll merge this or when so this will sit open for a bit.

**Pull request merged**